UNFLoader/device.cpp

```diff
@@ -223,11 +223,12 @@ void device_sendrom(char* rompath)
 
         // Open the ROM and get info about it 
         #ifndef LINUX
-            err = fopen_s(&file, rompath, "rb");
+        err = fopen_s(&file, rompath, "rb");
+        if (err != 0)
         #else
-            file = fopen(rompath, "rb");
+        file = fopen(rompath, "rb");
+        if (file == NULL)
         #endif
-        if (err != 0)

```

Comment:
Here I've moved error-checking into the conditional compilation since it depends if we're on *nix or not.